provider/spark.go

```diff
@@ -973,6 +1009,16 @@ func (s *S3Store) ResourceStream(key string) (chan []byte, error) {
 	return out, nil
 }
 
+func (s *S3Store) ResourceStreamCustom(key string, query string) (chan []byte, error) {
```

Comment:
Make this name more descriptive. Unsure what it does without looking through the code

Reply:
changed